pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -821,12 +835,64 @@ func loadBalancerServiceIsProgressing(ic *operatorv1.IngressController, service
 		}
 	}
 
+	if platform.Type == configv1.AWSPlatformType && eipAllocationsAWSEnabled {
+		var (
+			wantEIPAllocations, haveEIPAllocations []operatorv1.EIPAllocation
+			paramsFieldName                        string
+		)
+		switch getAWSLoadBalancerTypeInStatus(ic) {
+		case operatorv1.AWSNetworkLoadBalancer:
+			if nlbParams := getAWSNetworkLoadBalancerParametersInSpec(ic); nlbParams != nil {
+				wantEIPAllocations = nlbParams.EIPAllocations
+			}
+			if nlbParams := getAWSNetworkLoadBalancerParametersInStatus(ic); nlbParams != nil {
+				haveEIPAllocations = nlbParams.EIPAllocations
+			}
+			paramsFieldName = "networkLoadBalancer"
+		}
+		if !awsEIPAllocationsEqual(wantEIPAllocations, haveEIPAllocations) {
+			// Generate JSON for the oc patch command as well as "pretty" json
+			// that will be used for a more human-readable error message.
+			haveEIPAllocationsPatchJson := convertAWSEIPAllocationsListToPatchJson(haveEIPAllocations, "null", "null")
+			haveEIPAllocationsPrettyJson := convertAWSEIPAllocationsListToPatchJson(haveEIPAllocations, "{}", "[]")
+			wantEIPAllocationsPrettyJson := convertAWSEIPAllocationsListToPatchJson(wantEIPAllocations, "{}", "[]")
+			changedMsg := fmt.Sprintf("The IngressController eipAllocations were changed from %q to %q.", haveEIPAllocationsPrettyJson, wantEIPAllocationsPrettyJson)
+			ocPatchRevertCmd := fmt.Sprintf("oc -n openshift-ingress-operator patch ingresscontrollers/%[1]s --type=merge --patch='{\"spec\":{\"endpointPublishingStrategy\":{\"type\":\"LoadBalancerService\",\"loadBalancer\":{\"providerParameters\":{\"type\":\"AWS\",\"aws\":{\"type\":\"%[2]s\",\"%[3]s\":{\"eipAllocations\":%[4]s}}}}}}}'", ic.Name, getAWSLoadBalancerTypeInStatus(ic), paramsFieldName, haveEIPAllocationsPatchJson)
+			err := fmt.Errorf("%[1]s  To effectuate this change, you must delete the service: `oc -n %[2]s delete svc/%[3]s`; the service load-balancer will then be deprovisioned and a new one created. This will most likely cause the new load-balancer to have a different host name and IP address and cause disruption. To return to the previous state, you can revert the change to the IngressController: `%[4]s`", changedMsg, service.Namespace, service.Name, ocPatchRevertCmd)
+			errs = append(errs, err)
+		}
+	}
+
 	errs = append(errs, loadBalancerSourceRangesAnnotationSet(service))
 	errs = append(errs, loadBalancerSourceRangesMatch(ic, service))
 
 	return kerrors.NewAggregate(errs)
 }
 
+// convertAWSEIPAllocationsListToPatchJson converts an AWSEIPAllocations object to a JSON formatted string
+// to build an oc patch command. It defaults nil or no eipAllocation values to emptyEIPAllocationValue
+// and empty slices to emptyEIPAllocationSliceValue.
+func convertAWSEIPAllocationsListToPatchJson(eipAllocations []operatorv1.EIPAllocation, emptyEIPAllocationValue, emptyEIPAllocationSliceValue string) string {
```

Comment:
I would just simplify and make both empty struct and empty slice the same (`null` for patch, `[]` for pretty output):
 
```suggestion
func convertAWSEIPAllocationsListToPatchJson(eipAllocations []operatorv1.EIPAllocation, emptyEIPAllocationValue string) string {
```